src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy

```diff
@@ -125,14 +173,829 @@ class BidAdjustmentSpec extends BaseSpec {
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "Final bid price should be adjusted"
+        assert response.seatbid.first.bid.first.price == getAdjustedPrice(originalPrice, ruleValue as BigDecimal, adjustmentType)
+        assert response.cur == bidResponse.cur
+
+        and: "Original bid price and currency should be presented in bid.ext"
+        verifyAll(response.seatbid.first.bid.first.ext) {
+            origbidcpm == originalPrice
+            origbidcur == bidResponse.cur
+        }
+
+        and: "Bidder request should contain default c"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.cur == [currency]
+
+        where:
+        adjustmentType | maxRuleValue                | mediaType        | bidRequest
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | BANNER           | BidRequest.defaultBidRequest
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | VIDEO_IN_STREAM  | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_STREAM }
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | VIDEO_OUT_STREAM | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_ARTICLE }
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | AUDIO            | BidRequest.defaultAudioRequest
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | NATIVE           | BidRequest.defaultNativeRequest
+        MULTIPLIER     | MAX_MULTIPLIER_ADJUST_VALUE | ANY              | BidRequest.defaultBidRequest
+
+        CPM            | MAX_CPM_ADJUST_VALUE        | BANNER           | BidRequest.defaultBidRequest
+        CPM            | MAX_CPM_ADJUST_VALUE        | VIDEO_IN_STREAM  | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_STREAM }
+        CPM            | MAX_CPM_ADJUST_VALUE        | VIDEO_OUT_STREAM | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_ARTICLE }
+        CPM            | MAX_CPM_ADJUST_VALUE        | AUDIO            | BidRequest.defaultAudioRequest
+        CPM            | MAX_CPM_ADJUST_VALUE        | NATIVE           | BidRequest.defaultNativeRequest
+        CPM            | MAX_CPM_ADJUST_VALUE        | ANY              | BidRequest.defaultBidRequest
+
+        STATIC         | MAX_STATIC_ADJUST_VALUE     | BANNER           | BidRequest.defaultBidRequest
+        STATIC         | MAX_STATIC_ADJUST_VALUE     | VIDEO_IN_STREAM  | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_STREAM }
+        STATIC         | MAX_STATIC_ADJUST_VALUE     | VIDEO_OUT_STREAM | BidRequest.defaultVideoRequest.tap { imp.first.video.placement = IN_ARTICLE }
+        STATIC         | MAX_STATIC_ADJUST_VALUE     | AUDIO            | BidRequest.defaultAudioRequest
+        STATIC         | MAX_STATIC_ADJUST_VALUE     | NATIVE           | BidRequest.defaultNativeRequest
+        STATIC         | MAX_CPM_ADJUST_VALUE        | ANY              | BidRequest.defaultBidRequest
+    }
```

Comment:
The last one should be
`STATIC         | MAX_STATIC_ADJUST_VALUE        | ANY              | BidRequest.defaultBidRequest`
Add just one test case with the average max rule value.